src/rospack.cpp

```diff
@@ -2061,7 +2061,9 @@ Rosstackage::writeCache()
     {
       FILE *cache = fopen(tmp_cache_path, "w");
 #else
+    mode_t mask = umask(S_IRUSR | S_IWUSR| S_IXUSR | S_IRWXG);
```

Comment:
Your description mentioned 0600. Is that not a desired mask? Why also add read, write, execute/search for the group?

Is there a reason not to use `S_IRWXU` as a shortcut for the first three flags?

Reply:
Good point I will change it to `S_IRWXU`, I think default 0600 should be sufficient, but I'm not sure if any process requires read, write, execute group permission for concurrency. If not I will remove it. Thank you for your response